### PR TITLE
[AIRFLOW-5632] Rename GCP operators tests

### PR DESCRIPTION
**Pull request by michalslowikowski00**
- changed class name **GceHook**->**ComputeEngineHook**
- changed imports
- added deprecation warning for old module **GceHook**

---

Make sure you have checked _all_ steps below.

### Jira
Task can be found here https://issues.apache.org/jira/browse/AIRFLOW-5632

### Description
Pull request information:
- changed class name **GceHook**->**ComputeEngineHook**
- changed imports
- added deprecation warning for old module **GceHook**

### Tests
- added unit tests for tests/test_deprecation_warning.py

### CODE REVIEW

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 1]
```diff
@@ -21,9 +21,14 @@
...
-warnings.warn(
-    "This module is deprecated. Please use `airflow.gcp.hooks.compute`.",
-    DeprecationWarning, stacklevel=2
-)
+
+class GceHook(ComputeEngineHook):
+
+    def __init__(self, *args, **kwargs):
+        warnings.warn(
+            "This module is deprecated. Please use `airflow.gcp.hooks.compute`.",
```
Comment (nuclearpinguin) [inline airflow/contrib/hooks/gcp_compute_hook.py]: The message should include also information about the new operator that should be used by the user.
```suggestion
            "This module and operator are deprecated. Please use `airflow.gcp.hooks.compute.ComputeEngineOperator`.",
```

File: tests/test_raise_deprecation_warning.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    @parameterized.expand(PATH_TO_CLASSES)
+    def test_if_class_throw_deprecation_warning(self, new_module, old_module):
+        old_module_class = self.get_class_from_path(old_module)
+        with self.assertWarns(DeprecationWarning):
+            with mock.patch("{}.__init__".format(new_module)):
+                old_module_class()
+
+    @parameterized.expand(PATH_TO_CLASSES)
+    def test_if_is_subclass(self, parent_class, sub_class):
```
Comment (nuclearpinguin) [inline tests/test_raise_deprecation_warning.py]:
```suggestion
    def test_if_is_subclass(self, parent_class_path  , sub_class_path):
```

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 1 above)
Comment (mik-laj) [inline airflow/contrib/hooks/gcp_compute_hook.py]:
```suggestion
            "This class is deprecated. Please use `airflow.gcp.hooks.compute.XXX`.",
```

File: tests/test_raise_deprecation_warning.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        path, _, class_name = path_to_class.rpartition(".")
+        module = importlib.import_module(path)
+        class_ = getattr(module, class_name)
+        return class_
+
+    @parameterized.expand(PATH_TO_CLASSES)
+    def test_if_class_throw_deprecation_warning(self, new_module, old_module):
+        old_module_class = self.get_class_from_path(old_module)
+        with self.assertWarns(DeprecationWarning):
+            with mock.patch("{}.__init__".format(new_module)):
```
Comment (nuclearpinguin) [inline tests/test_raise_deprecation_warning.py]:
```suggestion
            with mock.patch("{}.__init__".format(new_module)) as init_mock:
```

File: tests/test_raise_deprecation_warning.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                old_module_class()
```
Comment (nuclearpinguin) [inline tests/test_raise_deprecation_warning.py]:
```suggestion
                old_module_class()
                init_mock.assert_called_once_with()
```
I think it’s worth to check if the mock was called.

File: tests/test_raise_deprecation_warning.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+# specific language governing permissions and limitations
+# under the License.
+import importlib
+import unittest
+from unittest import mock
+
+from parameterized import parameterized
+
+
+class TestRaiseDeprecationWarning(unittest.TestCase):
```
Comment (michalslowikowski00) [inline tests/test_raise_deprecation_warning.py]: Change test class name.

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 6]
```diff
@@ -16,14 +16,30 @@
...
+
+class GceHook(ComputeEngineHook):
+    """
+    This module is deprecated.
+    Please use `airflow.gcp.hooks.compute.ComputeEngineOperator`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        warnings.warn(
+            """These class and operator are deprecated.
```
Comment (nuclearpinguin) [inline airflow/contrib/hooks/gcp_compute_hook.py]:
```suggestion
            """This class is deprecated.
```

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 7]
```diff
@@ -16,14 +16,30 @@
...
+            Please use `airflow.gcp.hooks.compute.ComputeEngineOperator`.""",
```
Comment (nuclearpinguin) [inline airflow/contrib/hooks/gcp_compute_hook.py]:
```suggestion
            Please use `airflow.gcp.hooks.compute. ComputeEngineHook `.""",
```

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 8]
```diff
@@ -16,14 +16,30 @@
...
 
 import warnings
 
 # pylint: disable=unused-import
-from airflow.gcp.hooks.compute import GceHook, GceOperationStatus  # noqa
+from airflow.gcp.hooks.compute import ComputeEngineHook  # noqa
 
 warnings.warn(
-    "This module is deprecated. Please use `airflow.gcp.hooks.compute`.",
+    "This module is deprecated. airflow.gcp.hooks.compute.ComputeEngineOperator`",
```
Comment (nuclearpinguin) [inline airflow/contrib/hooks/gcp_compute_hook.py]:
```suggestion
    "This module is deprecated. airflow.gcp.hooks.compute. ComputeEngineHook `",
```

File: airflow/contrib/hooks/gcp_compute_hook.py [hunk 9]
```diff
@@ -16,14 +16,30 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-"""This module is deprecated. Please use `airflow.gcp.hooks.compute`."""
+"""This module is deprecated. Please use `airflow.gcp.hooks.compute.ComputeEngineOperator`."""
```
Comment (nuclearpinguin) [inline airflow/contrib/hooks/gcp_compute_hook.py]:
```suggestion
"""This module is deprecated. Please use `airflow.gcp.hooks.compute. ComputeEngineHook `."""
```

File: tests/test_core_to_contrib.py [hunk 10]
```diff
@@ -16,787 +16,819 @@
...
+        module = importlib.import_module(path)
+        class_ = getattr(module, class_name)
+        return class_
+
+    @parameterized.expand(MODULE)
+    def test_is_class_deprecated(self, new_module, old_module):
+        old_module_class = self.get_class_from_path(old_module)
+        with self.assertWarns(DeprecationWarning):
+            with mock.patch("{}.__init__".format(new_module)) as init_mock:
+                old_module_class()
```
Comment (nuclearpinguin) [inline tests/test_core_to_contrib.py]: We want to assert that the deprecation message includes information about new path to class.

File: tests/test_core_to_contrib.py [hunk 11]
```diff
@@ -16,787 +16,819 @@
...
+                init_mock.assert_called_once_with()
+
+    @parameterized.expand(MODULE)
+    def test_is_module_subclass(self, parent_class_path, sub_class_path):
```
Comment (nuclearpinguin) [inline tests/test_core_to_contrib.py]:
```suggestion
    def test_is_subclass(self, parent_class_path, sub_class_path):
```

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 9 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Right, thx.

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 6 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done.

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 7 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 6 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 8 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: airflow/contrib/hooks/gcp_compute_hook.py (hunk 9 above)
Comment (michalslowikowski00) [reply airflow/contrib/hooks/gcp_compute_hook.py]: Done

File: tests/test_core_to_contrib.py (hunk 11 above)
Comment (michalslowikowski00) [reply tests/test_core_to_contrib.py]: Done.

File: tests/test_core_to_contrib.py [hunk 12]
```diff
@@ -16,787 +16,821 @@
...
+        module = importlib.import_module(path)
+        class_ = getattr(module, class_name)
+        return class_
+
+    @parameterized.expand(MODULE)
+    def test_is_class_deprecated(self, new_module, old_module):
+        deprecation_warning_msg = "This class is deprecated."
+        old_module_class = self.get_class_from_path(old_module)
+        with self.assertWarnsRegex(DeprecationWarning, deprecation_warning_msg) as wrn:
+            with mock.patch("{}.__init__".format(new_module)) as init_mock:
```
Comment (nuclearpinguin) [inline tests/test_core_to_contrib.py]:
```suggestion
            old_module_path = old_module.rpartition(".")[0]
            new_class_name = new_module.rpartirion(",")[-1]
            mock_path = old_module_path + new_class_name
            with mock.patch("{}.__init__".format(mock_path)) as init_mock:
```

File: tests/test_core_to_contrib.py [hunk 13]
```diff
@@ -16,787 +16,821 @@
...
 import importlib
-from unittest import TestCase
+from unittest import TestCase, mock
 
 from parameterized import parameterized
 
 
 class TestMovingCoreToContrib(TestCase):
+
+    MODULE = [
```
Comment (nuclearpinguin) [inline tests/test_core_to_contrib.py]:
```suggestion
    HOOKS = [
```

File: tests/test_core_to_contrib.py (hunk 12 above)
Comment (nuclearpinguin) [inline tests/test_core_to_contrib.py]:
```suggestion
            with mock.patch("{}.__init__".format(new_module)) as init_mock:
                init_mock.return_value = None
```

Comment (michalslowikowski00) [reply tests/test_core_to_contrib.py]: Done

File: tests/test_core_to_contrib.py (hunk 10 above)
Comment (michalslowikowski00) [reply tests/test_core_to_contrib.py]: Done

File: tests/test_core_to_contrib.py (hunk 13 above)
Comment (michalslowikowski00) [reply tests/test_core_to_contrib.py]: Done

**Pull request closed without merging**